astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -362,6 +365,89 @@ def test_astype(self):
             int32q = self.q.astype("int32")
             assert_array_equal(np.astype(int32q, "int32"), int32q)
 
+    @needs_array_function
+    @pytest.mark.parametrize("angle_cls", [u.Quantity, Angle])
+    @pytest.mark.parametrize(
+        "args, kwargs, expected",
+        [
+            pytest.param((1 * u.radian,), {}, np.arange(1), id="pos: stop"),
+            pytest.param(
+                (0 * u.radian, 1 * u.degree), {}, np.arange(1), id="pos: start, stop"
+            ),
+            pytest.param(
+                (0 * u.radian, 1 * u.degree, 1 * u.arcsec),
+                {},
+                np.arange(ARCSEC_PER_DEGREE),
+                id="pos: start, stop, step",
+            ),
+            pytest.param(
+                (0 * u.radian, 1 * u.degree),
+                {"step": 1 * u.arcsec},
+                np.arange(ARCSEC_PER_DEGREE),
+                id="pos: start, stop; kw: step",
+            ),
+        ],
+    )
+    def test_arange(self, angle_cls, args, kwargs, expected):
+        arr = np.arange(
+            *args,
+            **kwargs,
+            like=angle_cls([], u.rad),
+        )
+        assert type(arr) is angle_cls
+        assert arr.unit == u.radian
+        np.testing.assert_array_almost_equal(arr.to_value(u.arcsec), expected)
+
+    @needs_array_function
+    @pytest.mark.parametrize(
+        "args, kwargs, expected_msg",
+        [
+            pytest.param(
+                (10 * u.s,),
+                {"stop": 5 * u.s},
+                "argument for arange() given by name ('stop') and position (position 1)",
+                id="1 posarg, 1 redundant kwarg",
+            ),
+            pytest.param(
+                (0 * u.s, 10 * u.s),
+                {"stop": 5 * u.s},
+                "argument for arange() given by name ('stop') and position (position 1)",
+                id="2 posargs, 1 redundant kwarg",
+            ),
+            pytest.param(
+                (0 * u.s, 10 * u.s, 2 * u.s),
+                {"stop": 5 * u.s},
+                "argument for arange() given by name ('stop') and position (position 1)",
+                id="3 posargs, 1 redundant kwarg",
+            ),
+            pytest.param(
+                (0 * u.s, 10 * u.s, 2 * u.s, "float64"),
+                {"stop": 5 * u.s},
+                "argument for arange() given by name ('stop') and position (position 1)",
+                id="4 posargs, 1 redundant kwarg",
+            ),
```

Comment:
note to reviewers: I know this test suite looks redudant, but it helped me discover that only one of them (the first one) wasn't passing without custom logic (which is probably a bug upstream). It's also nowhere near complete: there would be 11 other cases (if my math is right) to test for here, but I'm drawing *some* line between thorough testing and maintainability.

Reply:
I need to revisit this. I might not be relevant anymore (and I'm not convinced there's an upstream bug anymore)

Reply:
indeed I now think something was deeply wrong with my initial design. The tests that these comments were attached to are now removed because they ended up testing implementation details outside our control from *numpy*.